src/internal/components/configurationControls/axis/NameSubsection.tsx

```diff
@@ -0,0 +1,52 @@
+// (C) 2019 GoodData Corporation
+import get = require("lodash/get");
+import * as React from "react";
+import { InjectedIntlProps, injectIntl } from "react-intl";
+import ConfigSubsection from "../../configurationControls/ConfigSubsection";
+import NamePositionControl from "./NamePositionControl";
+import { IConfigItemSubsection } from "../../../interfaces/ConfigurationPanel";
+import { IVisualizationProperties } from "../../../interfaces/Visualization";
+
+class NameSubsection extends React.PureComponent<IConfigItemSubsection & InjectedIntlProps, {}> {
+    public render() {
+        const { axisVisible, axisNameVisible } = this.getControlProperties();
+        const { axis, properties, pushData, disabled, configPanelDisabled } = this.props;
+
+        return (
+            <ConfigSubsection
+                title="properties.axis.name"
+                valuePath={`${axis}.name.visible`}
+                properties={properties}
+                pushData={pushData}
+                canBeToggled={true}
+                toggledOn={axisNameVisible}
+                toggleDisabled={disabled || !axisVisible}
+                showDisabledMessage={!configPanelDisabled && disabled}
+            >
+                <NamePositionControl
+                    disabled={disabled}
+                    configPanelDisabled={configPanelDisabled}
+                    axis={axis}
+                    properties={properties}
+                    pushData={pushData}
+                />
+            </ConfigSubsection>
+        );
+    }
+
+    private getControlProperties(): IVisualizationProperties {
+        const axisVisible: boolean = get(this.props, `properties.controls.${this.props.axis}.visible`, true);
```

Comment:
we can cache this 
`const controlsAxis = get(this.props, "properties.controls.${this.props.axis}", {});`

Reply:
Fixed